### PR TITLE
RFC 2582: fix implicit auto-deref of raw pointers

### PR DESCRIPTION
**Pull request by RalfJung**
The code as given wouldn't even compile.

### CODE REVIEW

File: text/2582-raw-reference-mir-operator.md [hunk 1]
```diff
@@ -198,14 +198,14 @@ This has the side-effect of being able to entirely remove reference-to-pointer-*
...
-let field: *mut Field = &mut x.field;
+let field: *mut Field = &mut (*x).field;
 ```
 
 The lint as described in this RFC would nudge people to instead write
 
 ```rust
 let x: *mut Struct = NonNull::dangling().as_ptr();
-let field: *mut Field = &raw mut x.field;
+let field: *mut Field = &raw mut (x*).field;
```
Comment (SimonSapin) [inline text/2582-raw-reference-mir-operator.md:208]:
```suggestion
let field: *mut Field = &raw mut (*x).field;
```

Comment (RalfJung) [reply text/2582-raw-reference-mir-operator.md:208]: *oops* thanks!

**Comment by RalfJung** [comment]
Who'd usually review such editorial changes to existing RFCs? @nikomatsakis?

**Comment by pickfire** [comment]
Or maybe we should change the compiler such that it runs? I think the original code looks more ergonomic except that it doesn't work now.

**Comment by RalfJung** [comment]
@pickfire raw pointer very deliberately do not auto-deref as dereferencing a raw pointer needs to take great care to avoid causing [undefined behavior](https://doc.rust-lang.org/reference/behavior-considered-undefined.html).

**Review by ehuss — APPROVED** [review]

**Pull request merged**